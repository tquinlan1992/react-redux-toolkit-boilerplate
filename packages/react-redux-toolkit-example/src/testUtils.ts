import { RootState, extraArgsConnectProps } from "./store";
import {
  createTestComponent,
  mockHooks,
} from "@tquinlan1992/react-redux-toolkit/dist/testUtils";

const testUtilsExtraArgs = {
  ...extraArgsConnectProps,
  extraArgsMapHooks: {
    mockHooks: mockHooks(extraArgsConnectProps.extraArgsHooks.hooks),
  },
};

export const testComponent = createTestComponent<
  RootState,
  typeof testUtilsExtraArgs
>(testUtilsExtraArgs);
