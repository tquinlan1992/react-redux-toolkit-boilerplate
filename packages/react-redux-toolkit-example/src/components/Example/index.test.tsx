import Example from '.';
import { testComponent } from '../../testUtils';

testComponent('Example', Example, {
  mapState: ({ testMapState }) => {
    it('should mapstate', () => {
      testMapState(
        {
          example: {
            input1: 'input1',
            input2: 'input2',
          },
        },
        {
          input1: 'input1',
          input2: 'input2',
          hookInput: 'hookInput',
        },
      );
    });
  },
  mapHooks: ({ testMapHooks, testHookCalled, mockHooks }) => {
    it('should map hooks', () => {
      const state = { hookInput: 'hookInput', input1: 'input1', input2: 'input2' };
      testMapHooks({
        given: {
          state: state,
          hooks: mockHooks,
        },
        expect: {
          state,
          formDefaultValues: {
            username: 'initial username',
          },
        },
        calledWith: () => ({
          hooks: {
            hook1: testHookCalled(mockHooks.hook1, 'hookInput'),
          },
        }),
      });
    });
  },
  component: ({ testComponent }) => {
    it('props 1', () => {
      testComponent({
        state: {
          input1: 'input1',
          input2: 'input2',
          hookInput: 'hookInput',
        },
        hooks: {
          hook1: 'hook1',
        },
      });
    });
  },
});
