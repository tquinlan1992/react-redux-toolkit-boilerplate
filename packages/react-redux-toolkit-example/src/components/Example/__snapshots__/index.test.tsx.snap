// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example renders props 1 1`] = `
<React.Fragment>
  <h1>
    Example of redux toolkit
  </h1>
  <h1>
    Hooks results: 
    hook1
  </h1>
  <p>
    username input error: 
    username errors
  </p>
  <ForwardRef(Input)
    placeholder="username"
  />
  <ForwardRef(Button)>
    Handle React Form Submit
  </ForwardRef(Button)>
  <Memo(NoParams) />
  <ForwardRef(Button)
    onClick={[Function]}
  >
    Components.Button
  </ForwardRef(Button)>
  <div>
    <ForwardRef(Input)
      onChange={[Function]}
      placeholder="input1"
      value="input1"
    />
  </div>
  <div>
    <ForwardRef(Input)
      onChange={[Function]}
      placeholder="input2"
      value="input2"
    />
  </div>
  <div>
    <ForwardRef(Button)
      onClick="actions.resetAllInputs"
    >
      Reset All Input
    </ForwardRef(Button)>
    <ForwardRef(Button)
      onClick="actions.resetInput1"
    >
      Reset Input 1
    </ForwardRef(Button)>
    <ForwardRef(Button)
      onClick="actions.resetInput2"
    >
      Reset Input 2
    </ForwardRef(Button)>
    <ForwardRef(Button)
      onClick="actions.setInput1"
    >
      Set Input 1
    </ForwardRef(Button)>
    <ForwardRef(Button)
      onClick="actions.setInput2"
    >
      Set Input 2
    </ForwardRef(Button)>
    <ForwardRef(Button)
      onClick="actions.setAllInputs"
    >
      Set All Input
    </ForwardRef(Button)>
    <ForwardRef(Button)
      onClick="actions.setFromThunk"
    >
      Set From Thunk
    </ForwardRef(Button)>
    <Connected Hooks Component />
  </div>
</React.Fragment>
`;
